src/linux/mod.rs

```diff
@@ -273,6 +286,10 @@ impl<'a> SharedLibrary<'a> {
         F: FnMut(&Self) -> C,
         C: Into<IterationControl>,
     {
+        #[cfg(target_os = "android")]
```

Comment:
`dl_iterate_phdr` on android under api 27(8.1) will pass a zero-filled `dl_phdr_info` struct before any real shared library info, we should ignore it.